eng/common/scripts/ChangeLog-Operations.ps1

```diff
@@ -121,10 +122,88 @@ function Confirm-ChangeLogEntry {
   return $true
 }
 
-function Set-TestChangeLog($TestVersion, $changeLogFile, $ReleaseEntry) {
-  Set-Content -Path $changeLogFile -Value @"
-# Release History
-## $TestVersion ($(Get-Date -f "yyyy-MM-dd"))
-- $ReleaseEntry
-"@
+function New-ChangeLogEntry {
+  param (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [String]$Version,
+    [String]$Status=$UNRELEASED_TAG,
+    [String]$Title,
+    [String[]]$Content,
+    [String]$IgnoreInvalids=$True
+  )
+
+  # Validate RelaseStatus
+  $Status = $Status.Trim().Trim("()")
+  if ($Status -ne "Unreleased") {
+    try {
+      $Status = ([DateTime]$Status).ToString($DATE_FORMAT)
+    }
+    catch {
+        LogWarning "Invalid date [ $Status ] passed as status for Version [$Version]. Please use a valid date in the format '$DATE_FORMAT' or use '$UNRELEASED_TAG'"
+        if (!$IgnoreInvalids) { exit 1 }
```

Comment:
This pattern of ignoreInvalids is a little odd. Instead I would suggest we simply return $null and let the caller decide what to do.

Reply:
Well, this is going to be a problem. Right now not all entries in our changelogs have correct version. I was trying to have a way to detect bad version yet no have to break the process when reading ChangeLog entries.